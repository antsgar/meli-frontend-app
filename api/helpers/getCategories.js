const getCategories = ({ path_from_root }) => {
    return path_from_root.map(({ name }) => name);
};

module.exports = getCategories;
