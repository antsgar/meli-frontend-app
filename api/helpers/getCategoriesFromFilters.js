const getCategories = require("./getCategories");

const getCategoriesFromFilters = (filters) => {
    const categoryFilter = filters.find(
        ({ id, values }) => id === "category" && values.length > 0
    );
    return categoryFilter ? getCategories(categoryFilter.values[0]) : [];
};

module.exports = getCategoriesFromFilters;
