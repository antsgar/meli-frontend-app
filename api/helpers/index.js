const getCategories = require("./getCategories");
const getCategoriesFromFilters = require("./getCategoriesFromFilters");
const getItem = require("./getItem");
const getItems = require("./getItems");

module.exports = {
    getCategories,
    getCategoriesFromFilters,
    getItem,
    getItems,
};
