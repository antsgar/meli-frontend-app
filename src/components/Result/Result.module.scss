@import "../../styles/mixins";
@import "../../styles/vars";

.hoverable {
    background-color: white;

    &:first-child {
        border-radius: 4px 4px 0 0;
    }

    &:last-child {
        border-radius: 0 0 4px 4px;
    }

    &:hover {
        opacity: 0.7;
        cursor: pointer;
    }
    

    &:not(:last-child) {
        .container {
            border-bottom: 1px solid $border-color-default;
        }
    }
}

.container {
    @include gridWrapper(10);
    margin: 0 $spacing-medium;
    padding: $spacing-medium 0;
}

.content {
    @include gridContent(1, 11);
    display: flex;

    @include desktop {
        @include gridContent(1, 10);
    }
}

.picture {
    height: 90px;
    width: 90px;
    border-radius: 4px;
    align-self: center;

    @include desktop {
        height: 180px;
        width: 180px;
    }
}

.productInfoContainer {
    flex: 1;
    margin: $spacing-medium 0 0 $spacing-medium;
}

.priceAndStateContainer {
    @include desktop {
        display: flex;
        align-items: baseline;
    }
}

.priceAndShipping {
    flex: 1;
    font-size: 24px;
    color: $text-color-dark;
    display: flex;
    align-items: center;
}

.freeShippingIcon {
    margin-left: $spacing-small;
    height: 20px;
}

.stateName {
    font-size: 12px;
    color: $text-color-default;
    margin-top: $spacing-medium;
}

.title {
    font-size: 18px;
    color: $text-color-dark;
    margin-top: $spacing-large;
    margin-right: $spacing-large;
}
