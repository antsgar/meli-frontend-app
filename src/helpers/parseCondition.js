const parseCondition = (condition) => {
    switch (condition) {
        case "new":
            return "Nuevo";
        case "used":
            return "Usado";
        default:
            return "Nuevo";
    }
};

export default parseCondition;
