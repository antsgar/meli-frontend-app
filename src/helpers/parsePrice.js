const parsePrice = (price) => {
    return new Intl.NumberFormat("es-AR", {
        style: "currency",
        currency: price.currency,
        minimumFractionDigits: 0,
    }).format(Math.round(price.amount));
};

export default parsePrice;
