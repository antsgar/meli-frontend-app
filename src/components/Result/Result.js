import { useRouter } from "next/router";
import { parsePrice } from "../../helpers";
import styles from "./Result.module.scss";

const Result = ({ result }) => {
    const router = useRouter();

    const {
        id,
        picture,
        price,
        title,
        state_name: stateName,
        free_shipping: freeShipping,
    } = result;

    const openItemDetail = () => {
        router.push("/items/[id]", `/items/${id}`);
    };

    return (
        <div
            className={styles.hoverable}
            data-test="result-item"
            data-id={id}
            onClick={openItemDetail}
        >
            <div className={styles.container}>
                <div className={styles.content}>
                    <img
                        className={styles.picture}
                        src={picture}
                        alt="Product picture"
                    ></img>
                    <div className={styles.productInfoContainer}>
                        <div className={styles.priceAndStateContainer}>
                            <div className={styles.priceAndShipping}>
                                {parsePrice(price)}
                                {freeShipping && (
                                    <img
                                        className={styles.freeShippingIcon}
                                        src="/ic_shipping@2x.png.png"
                                        alt="Free shipping icon"
                                    ></img>
                                )}
                            </div>
                            <div className={styles.stateName}>{stateName}</div>
                        </div>
                        <div className={styles.title}>{title}</div>
                    </div>
                </div>
            </div>
        </div>
    );
};

export default Result;
