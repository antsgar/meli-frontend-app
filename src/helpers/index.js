import parseCondition from "./parseCondition";
import parsePrice from "./parsePrice";

export { parseCondition, parsePrice };
