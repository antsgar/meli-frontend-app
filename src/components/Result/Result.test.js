import React from "react";
import { mount } from "enzyme";
import Result from "./Result";
import { items } from "../../fixtures";
import { parsePrice } from "../../helpers";

const [result] = items;
let wrapper;

beforeEach(() => {
    wrapper = mount(<Result result={result} />);
});

describe("Result", () => {
    it("should render component", () => {
        expect(wrapper).toMatchSnapshot();
    });

    it("should display product information", () => {
        const { title, price, free_shipping, state_name } = result;

        expect(wrapper.find(".picture").exists()).toEqual(true);
        expect(wrapper.find(".priceAndShipping").text()).toEqual(
            parsePrice(price)
        );
        expect(wrapper.find(".freeShippingIcon").exists()).toEqual(
            free_shipping
        );
        expect(wrapper.find(".stateName").text()).toEqual(state_name);
    });
});
