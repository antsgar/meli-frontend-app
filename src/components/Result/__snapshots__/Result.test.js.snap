// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Result should render component 1`] = `
<Result
  result={
    Object {
      "condition": "new",
      "free_shipping": true,
      "id": "MLA860381954",
      "picture": "http://mla-s1-p.mlstatic.com/885508-MLA32445302787_102019-I.jpg",
      "price": Object {
        "amount": 166799,
        "currency": "ARS",
        "decimals": null,
      },
      "state_name": "Capital Federal",
      "title": "iPhone 11 64 Gb Negro 4 Gb Ram",
    }
  }
>
  <div
    className="hoverable"
    data-id="MLA860381954"
    data-test="result-item"
    onClick={[Function]}
  >
    <div
      className="container"
    >
      <div
        className="content"
      >
        <img
          alt="Product picture"
          className="picture"
          src="http://mla-s1-p.mlstatic.com/885508-MLA32445302787_102019-I.jpg"
        />
        <div
          className="productInfoContainer"
        >
          <div
            className="priceAndStateContainer"
          >
            <div
              className="priceAndShipping"
            >
              ARSÂ 166,799
              <img
                alt="Free shipping icon"
                className="freeShippingIcon"
                src="./ic_shipping@2x.png.png"
              />
            </div>
            <div
              className="stateName"
            >
              Capital Federal
            </div>
          </div>
          <div
            className="title"
          >
            iPhone 11 64 Gb Negro 4 Gb Ram
          </div>
        </div>
      </div>
    </div>
  </div>
</Result>
`;
